main.go

```diff
@@ -95,6 +96,9 @@ func main() {
 	flag.IntVar(&machineDeploymentConcurrency, "machinedeployment-concurrency", 10,
 		"Number of machine deployments to process simultaneously")
 
+	flag.BoolVar(&kubeadmEnabled, "kubeadm-enabled", true,
+		"Whether or not to deploy the kubeadm bootstrap and controlplane components")
```

Comment:
a non-blocking comment, but the list of CLI flags here suggests a component-configuration at some point.
if this was a configuration format the existence of a kubeadm structure (on by default) would have suggested that kubeadm usage is desired and all the kubeadm specifics could be fields of the same structure.

Reply:
Agree, but I think we should likely create a separate issue for that.